dao/donation.py

```diff
@@ -1,12 +1,72 @@
 import datetime
 
 from config import db
+from dao.mixin import OutputMixin
 
 
-class Donation(db.Model):
+class Donation(OutputMixin, db.Model):
+    RELATIONSHIPS_TO_DICT = True
+    DONATION_REQUIRED_PARAMS = ['supplyName', 'quantity', 'unit', 'uid']
+
     did = db.Column(db.Integer, primary_key=True)
     supplyName = db.Column(db.String(30), nullable=False)
     quantity = db.Column(db.Integer, default=1)
     createdAt = db.Column(db.DateTime, default=datetime.datetime.utcnow())
     unit = db.Column(db.String(30), nullable=False)
     uid = db.Column(db.Integer, db.ForeignKey('user.uid'), nullable=False)
+    requests = db.relationship('Request', backref=db.backref('donation', lazy='subquery'), lazy=True)
+
+    def __init__(self, **kwargs):
+        self.supplyName = kwargs.get('supplyName')
+        self.quantity = kwargs.get('quantity')
+        self.unit = kwargs.get('unit')
+        self.uid = kwargs.get('uid')
+
+    def __repr__(self):
+        return self.supplyName
+
+    @property
+    def pk(self):
+        return self.did
+
+    @staticmethod
+    def get_all_donations():
+        return Donation.query.all()
+
+    @staticmethod
+    def get_all_donations_without_request():
+        return db.session.query(Donation).filter(Donation.requests == None).all()
```

Comment:
I think this query can be simplified more like the one in `get_available_donations`. Since the requests field of the donation is specified to bring all the requests related in a subquery.